pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -154,9 +161,18 @@ func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
 		// cache may be stale and we don't want to overwrite a prior update
 		// to status with this stale state.
 	} else if _, err := c.updateStatus(pr); err != nil {
-		c.Logger.Warn("Failed to update PipelineRun status", zap.Error(err))
+		message := "Failed to update PipelineRun status %s"
+		c.Logger.Warnf(message, zap.Error(err))
+		c.Recorder.Eventf(pr, corev1.EventTypeWarning, eventReasonFailed, message, zap.Error(err))
 		return err
 	}
+
+	if err == nil {
+		c.Recorder.Eventf(pr, corev1.EventTypeNormal, eventReasonSucceeded, "PipelineRun '5s' is done.", pr.Name)
+	} else {
+		c.Recorder.Eventf(pr, corev1.EventTypeWarning, eventReasonFailed,
+			"Failed to execute PipelineRun status %s", zap.Error(err))
```

Comment:
Similarly I think it would be better if msg is `Pipelinerun failed to reconcile` here. I am not fan of exposing errors in event msgs so I would not attach that. We should be logging all errors in controller instead of relying on event msgs.

Reply:
Agreed.  

I also want hear from other reviwer's opinion on logging error messages in events.  

Thank you so much for your great feedbacks!

Reply:
agreeing with @shashwathi :angel: